File: DSA_Questions/src/com/dsa_questions/Minimum_platform_required.java
package com.dsa_questions;

import java.util.*;

public class Minimum_platform_required {

	    
	    public static void main(String[] args){
	        
	        Scanner sc = new Scanner(System.in);
	        
	        int N = sc.nextInt();
	        
	        int[][] ari = new int[N][2];
	        
	        int[][] dept = new int[N][2];
	        
	        int[][] mat = new int[N*2][2];
	        
	        for(int i = 0;i < N;i++){
	            
	            String str = sc.next();
	            
	            if(str.length() == 5){
	                
	                str = "" + str.charAt(0) + str.charAt(1) + str.charAt(3) + str.charAt(4);
	                
	            }else{
	                
	                str = "" + str.charAt(0) + str.charAt(2) + str.charAt(3);
	                
	            }
	            
	            ari[i][0] = Integer.parseInt(str);
	            
	            ari[i][1] = 1;
	            
	            mat[i][0] = ari[i][0];
	            
	            mat[i][1] = ari[i][1];
	            
	        }
	        
	        
	        for(int i = 0;i < N;i++){
	            
	            String str = sc.next();
	            
	            if(str.length() == 5){
	                
	                str = "" + str.charAt(0) + str.charAt(1) + str.charAt(3) + str.charAt(4);
	                
	            }else{
	                
	                str = "" + str.charAt(0) + str.charAt(2) + str.charAt(3);
	                
	            }
	            
	            dept[i][0] = Integer.parseInt(str);
	            
	            dept[i][1] = -1;
	            
	            mat[i+N][0] = dept[i][0];
	            
	            mat[i+N][1] = dept[i][1];
	        }
	        
	        Comparator<int[]> comp = (o1,o2) -> o1[0] > o2[0] ? 1 : -1;
	        
	        Arrays.sort(mat, comp);
	        
	        int sum = 0;
	        
	        int max = 0;
	        
	        for(int i = 0;i < N*2;i++){
	            
	            sum += mat[i][1];
	            
	            max = Math.max(max, sum);

	        }
	        
	        System.out.println(max);
	        
	    }
	    
	}
	

File: DSA_Questions_Python/Minimum_Platform_required.py
N = int(input())

ari = [[0, 1] for _ in range(N)]
dept = [[0, -1] for _ in range(N)]
mat = [[0, 0] for _ in range(N * 2)]

for i in range(N):
    str_ari = input()
    if len(str_ari) == 5:
        str_ari = str_ari[0] + str_ari[1] + str_ari[3] + str_ari[4]
    else:
        str_ari = str_ari[0] + str_ari[2] + str_ari[3]
    
    ari[i][0] = int(str_ari)
    mat[i][0] = ari[i][0]

for i in range(N):
    str_dept = input()
    if len(str_dept) == 5:
        str_dept = str_dept[0] + str_dept[1] + str_dept[3] + str_dept[4]
    else:
        str_dept = str_dept[0] + str_dept[2] + str_dept[3]

    dept[i][0] = int(str_dept)
    mat[i + N][0] = dept[i][0]

mat.sort(key=lambda x: x[0])

sum_ = 0
max_ = 0

for i in range(N * 2):
    sum_ += mat[i][1]
    max_ = max(max_, sum_)

print(max_)
